src/ert/config/ert_script.py

```diff
@@ -214,23 +213,24 @@ def loadScriptFromFile(
     @staticmethod
     def __findErtScriptImplementations(
         module: ModuleType,
-    ) -> Callable[[], ErtScript]:
-        result = []
+    ) -> type[ErtScript]:
+        result = None
         for _, member in inspect.getmembers(
             module,
             lambda member: inspect.isclass(member)
             and member.__module__ == module.__name__,
         ):
             if ErtScript in inspect.getmro(member):
-                result.append(member)
+                if result:
+                    raise ValueError(
+                        f"Module {module.__name__} contains more than one ErtScript"
+                    )
+                result = member
 
-        if len(result) == 0:
+        if result is None:
             raise ValueError(f"Module {module.__name__} does not contain an ErtScript!")
-        if len(result) > 1:
-            raise ValueError(
-                f"Module {module.__name__} contains more than one ErtScript"
-            )
-        return result[0]
+        assert issubclass(result, ErtScript)
```

Comment:
Don't think this is needed since we have already checked that it is in the mro

Reply:
Not really, but pyCharm complains